pkg/storage/fs/local/local.go

```diff
@@ -84,13 +87,26 @@ func (fs *localFS) removeRoot(np string) string {
 
 type localFS struct{ root string }
 
+func calcEtag(fi os.FileInfo) string {
+	stat, ok := fi.Sys().(*syscall.Stat_t)
+	h := md5.New()
+	binary.Write(h, binary.BigEndian, fi.ModTime().Unix())
+	if ok {
+		// take device and inode into account
+		binary.Write(h, binary.BigEndian, stat.Ino)
+		binary.Write(h, binary.BigEndian, stat.Dev)
+	}
+	binary.Write(h, binary.BigEndian, fi.Size())
+	return fmt.Sprintf(`"%x"`, h.Sum(nil))
+}
+
 func (fs *localFS) normalize(fi os.FileInfo, fn string) *storage.MD {
 	fn = fs.removeRoot(path.Join("/", fn))
 	md := &storage.MD{
 		ID:          "fileid-" + strings.TrimPrefix(fn, "/"),
 		Path:        fn,
 		IsDir:       fi.IsDir(),
-		Etag:        fmt.Sprintf("%d", fi.ModTime().Unix()),
+		Etag:        calcEtag(fi),
```

Comment:
etag calculation moved to a function that returns a quoted string and takes into account inode and dev if they are available